### ISSUE TITLE
[CLI] Fix dependency with @ledgerhq/hw-transport-node-hid v5.36.0

### ISSUE THREAD

**Issue by medhakothari** [celo-org/celo-monorepo#6274]
### Expected Behavior

Celocli should work with version 5.36.0 of the above package. 

### Current Behavior
The cli consumes [the @ledgerhq/hw-transport-node-hid package](https://www.npmjs.com/package/@ledgerhq/hw-transport-node-hid/), which recently went from 5.34.0 to 5.36.0. Something in this upgrade brought about the following Node error when trying to send transactions using the celocli:
`RangeError: Index out of range
Code: ERR_INDEX_OUT_OF_RANGE`
Temporary solution was to downgrade the ledger package. 

See [this discord discussion](https://discord.com/channels/600834479145353243/601046166620078100/788507609028231199) for more information.

**Comment by tanguyonlivelab** [comment]
I tried running "npm i @ledgerhq/hw-transport-node-hid@5.34.0" which I found here https://www.npmjs.com/package/@ledgerhq/hw-transport-node-hid/v/5.34.0 but still no success. Do you have another suggestion on how to tackle this issue?

**Comment by alonshavit** [comment]
To anyone having the same issue the temporary solution described [here](https://discord.com/channels/600834479145353243/601046166620078100/795944287463538688) solved the issue for me.

**Comment by zviadm** [comment]
Pasting workaround here, so you don't have to dig through Discord:

```
1) First downgrade ledgerhq package like described by @vires-in-numeris:

You need to downgrade the whole ledgerhq package (don't know where exactly the error comes from). I did the following if I remember correctly:
1) mkdir celo && cd celo
2) yarn add @celo/celocli
3) add "resolutions" to the package.json file
4) yarn install
here's my package.json
{
  "dependencies": {
    "@celo/celocli": "0.0.58"
  },
  "resolutions": {
    "@ledgerhq/hw-transport-node-hid": "5.34.0",
    "@ledgerhq/cryptoassets": "5.34.0",
    "@ledgerhq/devices": "5.34.0",
    "@ledgerhq/errors": "5.34.0",
    "@ledgerhq/hw-app-eth": "5.34.0",
    "@ledgerhq/hw-transport-node-hid-noevents": "5.34.0",
    "@ledgerhq/hw-transport": "5.34.0",
    "@ledgerhq/logs": "5.38.0"
  }
}

2) Then use celocli with "yarn run" in the local folder (then celocli does not use the global ledgerhq package, but the recently installed downgraded package):

e.g. yarn run celocli releasegold:transfer-dollars .....
```

This seems like a pretty major issue btw, and it has been >1 month since it was reported.

**Comment by medhakothari** [comment]
Thank you for all the helpful instructions and workarounds posted above!
After digging into the issue, I found that when we RLP encode Celo txns, the [chainID is on index 9](https://github.com/celo-org/celo-monorepo/blob/6148646a29e5e719df9a1b32d1580ea7e6e61be8/packages/sdk/wallets/wallet-base/src/signing-utils.ts#L103). But the @ledgerhq/hw-app-eth package, after the 5.36.0 version, gets the chainID [from index 6](https://github.com/LedgerHQ/ledgerjs/blob/ebd5e9c4d97cb3e83c6d90ab004d28dbac1d188f/packages/hw-app-eth/src/Eth.js#L208), since it is optimized for ethereum txns.

So for now, we're fixing the version of that dependency to 5.11.0; but longer term we plan on creating our own Celo-specific library for this. See the issue to track this [here](https://github.com/celo-org/celo-monorepo/issues/6495). 

I will be releasing an updated version of the CLI with this change included in the next few hours; will continue to update this issue. Edit: the fix is now available on version 1.0.0-beta2 of the CLI. You can download it using this command: `npm install -g @celo/celocli@1.0.0-beta2`
